core/src/autogluon/core/models/greedy_ensemble/ensemble_selection.py

```diff
@@ -176,6 +169,10 @@ def _fit(self, predictions: List[np.ndarray], labels: np.ndarray, time_limit=Non
             order.append(best)
             used_models.add(best)
 
+            ensemble_prediction *= s
+            ensemble_prediction += ensemble[-1]
+            ensemble_prediction /= s+1
```

Comment:
I wonder if this may lead to loss of numerical precision when we multiple by a large number `s` and then divide it back by `s+1`. Is this code equivalent to the following?
```python
ensemble_prediction =  ensemble_prediction * s / (s + 1) + ensemble[-1] / (s + 1)
```

Reply:
Good point, yes this should be equivalent. I will replace this part with your code.